x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.ts

```diff
@@ -7,75 +7,67 @@
 
 import { kea, MakeLogicType } from 'kea';
 
-import { EuiGlobalToastListToast as IToast } from '@elastic/eui';
+import type { NotificationsStart } from '@kbn/core-notifications-browser';
 
 import { KibanaLogic } from '../kibana';
 
 import { IFlashMessage } from './types';
 
 interface FlashMessagesValues {
+  historyListener: Function | null;
   messages: IFlashMessage[];
+  notifications: NotificationsStart;
   queuedMessages: IFlashMessage[];
-  toastMessages: IToast[];
-  historyListener: Function | null;
 }
 interface FlashMessagesActions {
   setFlashMessages(messages: IFlashMessage | IFlashMessage[]): { messages: IFlashMessage[] };
   clearFlashMessages(): void;
   setQueuedMessages(messages: IFlashMessage | IFlashMessage[]): { messages: IFlashMessage[] };
   clearQueuedMessages(): void;
-  addToastMessage(newToast: IToast): { newToast: IToast };
-  dismissToastMessage(removedToast: IToast): { removedToast: IToast };
-  clearToastMessages(): void;
   setHistoryListener(historyListener: Function): { historyListener: Function };
 }
 
+interface FlashMessagesLogicProps {
+  notifications: NotificationsStart;
+}
+
 const convertToArray = (messages: IFlashMessage | IFlashMessage[]) =>
   !Array.isArray(messages) ? [messages] : messages;
 
-export const FlashMessagesLogic = kea<MakeLogicType<FlashMessagesValues, FlashMessagesActions>>({
-  path: ['enterprise_search', 'flash_messages_logic'],
+export const FlashMessagesLogic = kea<
+  MakeLogicType<FlashMessagesValues, FlashMessagesActions, FlashMessagesLogicProps>
+>({
   actions: {
-    setFlashMessages: (messages) => ({ messages: convertToArray(messages) }),
     clearFlashMessages: () => null,
-    setQueuedMessages: (messages) => ({ messages: convertToArray(messages) }),
     clearQueuedMessages: () => null,
-    addToastMessage: (newToast) => ({ newToast }),
-    dismissToastMessage: (removedToast) => ({ removedToast }),
-    clearToastMessages: () => null,
+    setFlashMessages: (messages) => ({ messages: convertToArray(messages) }),
     setHistoryListener: (historyListener) => ({ historyListener }),
+    setQueuedMessages: (messages) => ({ messages: convertToArray(messages) }),
   },
-  reducers: {
+  path: ['enterprise_search', 'flash_messages_logic'],
+  reducers: ({ props }) => ({
+    historyListener: [
+      null,
+      {
+        setHistoryListener: (_, { historyListener }) => historyListener,
+      },
+    ],
     messages: [
       [],
       {
-        setFlashMessages: (_, { messages }) => messages,
         clearFlashMessages: () => [],
+        setFlashMessages: (_, { messages }) => messages,
       },
     ],
+    notifications: [props.notifications || {}, {}],
```

Comment:
I had to add the `|| {}` here to get an unrelated unit test to pass. It seems like there was a transitive dependency that was mounting this logic without the helper and this was undefined throwing a kea error.

This did not fill me with a lot of confidence :/ but I'm pretty sure this should always have a value at runtime since we mount this in `renderApp` and always pass the start object.

Reply:
Another option could be to introduce a logic specifically to hold the `notifications` and use it in `flashSuccessToast` & `flashErrorToast`

Reply:
As long as logics passed correctly at the time of mounting this logic, it would have a value (which it does in this case).
Even kea example has `props.defaultCounter || 0` as an example. https://v2.keajs.org/docs/guide/additional#props